### ISSUE TITLE
Cloudfront parameter to NOT wait for distribution to finish deploying

### ISSUE THREAD

**Issue by t-pascal** [terraform-providers/terraform-provider-aws#8175]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In #260 the feature was to change the default behavior for deploying cloudfront to wait until the distribution is completely deployed. This takes approximately 8-20 minutes. Unfortunately, we have a dozen cloudfront terraforms that we manage and our deployment pipline which used to complete in a few minutes now delays and waits an intolerable amount of time.

I feel like changing the default was a reasonable decision, but the option to restore the previous behavior (without locking our provider version) would be a welcome feature.

### New or Affected Resource(s)

* aws_cloudfront_distribution

### Potential Terraform Configuration

```hcl
resource "aws_cloudfront_distribution" "consumer_traffic" {
  aliases             = ["${split(",",var.alternate_domain_names)}"]
  comment             = "Consumer Frontend - ${var.distribution_name}"
  default_root_object = ""
  price_class         = "PriceClass_100"
  enabled             = true
  wait_for_deployment = false
…
}
```

### References

* #260

* https://github.com/hashicorp/terraform/issues/8749
* #7794

**Comment by bflad** [comment]
Hi @t-pascal 👋 Thanks for submitting this. Good news, support for a new argument to disable the behavior, `wait_for_deployment`, has been merged via #8116 and will release later this week with [version 2.5.0 of the Terraform AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#250-unreleased). Since this is already completed and just pending release I'm going to close this issue. 👍

**Comment by rtlong** [comment]
See also #8073
